packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -28,11 +32,29 @@ export class ProductStack extends cdk.Stack {
     });
 
     this._parentStack = findParentStack(scope);
-
+    this._productVersionDetails = new ProductVersionDetails();
     // this is the file name of the synthesized template file within the cloud assembly
     this.templateFile = `${cdk.Names.uniqueId(this)}.product.template.json`;
   }
 
+  /**
+   * Fetch the product version details.
+   *
+   * @internal
+   */
+  public _getProductVersionDetails(): ProductVersionDetails {
+    return this._productVersionDetails;
+  }
+
+  /**
+   * Set the parent product stack history
+   *
+   * @internal
+   */
+  public _setParentProductStackHistory(parentProductStackHistory: ProductStackHistory) {
+    return this._parentProductStackHistory = parentProductStackHistory;
+  }
+
```

Comment:
Please try to find a way to implement this feature *without* changing `ProductStack` (or convince me it's not possible).